app/assets/javascripts/comfortable_mexican_sofa/admin/modules/CategoryExpand.js

```diff
@@ -0,0 +1,76 @@
+define(['jquery','DoughBaseComponent'], function ($, DoughBaseComponent) {
+'use strict';
+
+  var defaultConfig = {
+    categoriesCharacterCount: 20
+  };
+  var CategoryExpandProto;
+
+  function CategoryExpand($el, config) {
+    CategoryExpand.baseConstructor.call(this, $el, config, defaultConfig);
+  }
+
+  DoughBaseComponent.extend(CategoryExpand);
+
+  CategoryExpand.componentName = 'CategoryExpand';
+
+  CategoryExpandProto = CategoryExpand.prototype;
+
+  CategoryExpandProto.init = function(initialised) {
+
+    this.addExpanders();
+    this.addListeners();
+
+    this._initialisedSuccess(initialised);
+    return this;
+
+  };
+
+  CategoryExpandProto.addListeners = function() {
+
+    $('#filter_form').on('ajax:complete', $.proxy(function() {
+      this.addExpanders();
+    }, this));
+
+  };
+
+  CategoryExpandProto.addExpanders = function() {
+
+    var self = this;
+
+    this.$el.find('.categories').each(function() {
+
+      var categories = $(this).text(),
+          categoryCount = categories.split(',').length - 1,
+          categoriesCharacterCount = self.config.categoriesCharacterCount;
+
+      if (categories.indexOf(',') >= 1) {
+        $(this).html(
+          categories.slice(0,categoriesCharacterCount)+
+          '<a href="#" class="more"> +' + categoryCount + ' see all</a>'+
+          '<span style="display:none;" class="allCategories">'+ categories.slice(categoriesCharacterCount ,categories.length)+' <a href="#" class="less">Hide</a></span>'
+        );
+      };
```

Comment:
It might be a nice idea to pull this out as a JS $template partial using say mustache, handlebars or lodash (whichever we have) and then we can then pass it parameters to more easily build up the categories list.

We can go through this tomorrow if you like.